sdk/cognitiveservices/azopenai/custom_client.go

```diff
@@ -0,0 +1,57 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package azopenai
+
+// this file contains handwritten additions to the generated code
+
+import (
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+)
+
+const (
+	clientName = "azopenai.Client"
+	apiVersion = "2022-12-01"
+	tokenScope = "https://cognitiveservices.azure.com/.default"
+)
+
+// Clients
+
+// ClientOptions contains optional settings for Client.
+type ClientOptions struct {
+	azcore.ClientOptions
+}
+
+// NewClient creates a new instance of Client with the specified values.
+//   - endpoint - OpenAI service endpoint
+//   - credential - used to authorize requests. Usually a credential from azidentity.
+//   - options - client options, pass nil to accept the default values.
+func NewClient(endpoint string, credential azcore.TokenCredential, options *ClientOptions) (*Client, error) {
+	if options == nil {
+		options = &ClientOptions{}
+	}
+	authPolicy := runtime.NewBearerTokenPolicy(credential, []string{tokenScope}, nil)
```

Comment:
We should probably add the cloud configuration logic here.

Reply:
Do we need to do something here? I thought the cloud configuration was strictly for ARM clients?

Reply:
Issue filed.